### PR TITLE
Fixed `compas_ghpython` is not importable in Rhino8

### PR DESCRIPTION
**Pull request by chenkasirer**
* Fixed `compas_ghpython` not importable in Rhino 8 CPython components, mostly by:
    * Avoiding `compas_rhino.drawing` imports within `compas_ghpython`
    * Modifying the plugin discovery logic to catch a new type of exception.
* Retired deprecated module `compas_ghpython.utilities`.

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [x] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [ ] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [ ] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas/plugins.py [hunk 1]
```diff
@@ -33,6 +35,17 @@
     "PluginDefaultNotAvailableError",
 ]
 
+if compas.RHINO:
+    import System  # type: ignore  # pragma: no cover
```
Comment (tomvanmele) [inline src/compas/plugins.py]: i would not add these pragma annotations. we don't really care about the coverage notifications, so i would also not pollute the codebase to reduce their frequency...

File: src/compas/plugins.py [hunk 2]
```diff
@@ -33,6 +35,17 @@
...
+
+    DotNetException = System.Exception  # pragma: no cover
+else:
+
+    class DummyDotNetException(BaseException):
+        pass
+
+    DotNetException = DummyDotNetException
```
Comment (tomvanmele) [inline src/compas/plugins.py]: why is this done in two steps?

File: src/compas/plugins.py [hunk 3]
```diff
@@ -401,15 +414,15 @@ def try_import(self, module_name):
...
         try:
             module = __import__(module_name, fromlist=["__name__"], level=0)
             self._cache[module_name] = True
 
         # There are two types of possible failure modes:
         # 1) cannot be imported, or
         # 2) is a python 3 module and we're in IPY, which causes a SyntaxError
-        except (ImportError, SyntaxError):
+        # 3) in Rhino8 we may get a nasty DotNet Exception when trying to import a module
+        except (ImportError, SyntaxError, DotNetException):
```
Comment (tomvanmele) [inline src/compas/plugins.py:425]: does this mean, when not in Rhino, we accept `ImportError`, `SyntaxError` AND `BaseException`? does this mean we accept ALL exceptions?

File: src/compas_ghpython/scene/meshobject.py [hunk 4]
```diff
@@ -147,11 +146,21 @@ def draw_faces(self):
...
+
+    @staticmethod
+    def _create_ngon(vertex_count):
+        if vertex_count < 3:
+            return
+        if vertex_count == 3:
+            return [0, 1, 2]
+        if vertex_count == 4:
+            return [0, 1, 2, 3]
+        return list(range(vertex_count))
```
Comment (tomvanmele) [inline src/compas_ghpython/scene/meshobject.py]: no reason to attach this to the object

File: src/compas_ghpython/scene/volmeshobject.py [hunk 5]
```diff
@@ -115,3 +114,13 @@ def draw_cells(self):
...
+
+    @staticmethod
+    def _create_ngon(vertex_count):
+        if vertex_count < 3:
+            return
+        if vertex_count == 3:
+            return [0, 1, 2]
+        if vertex_count == 4:
+            return [0, 1, 2, 3]
+        return list(range(vertex_count))
```
Comment (tomvanmele) [inline src/compas_ghpython/scene/volmeshobject.py]: same

**Review by tomvanmele — COMMENTED** [review]
other than a few questions, i think this is great!

File: src/compas/plugins.py (hunk 2 above)
Comment (chenkasirer) [reply src/compas/plugins.py]: what do you mean two steps? why I'm not using `BaseException` directly?

File: src/compas/plugins.py (hunk 3 above)
Comment (chenkasirer) [reply src/compas/plugins.py:425]: outside Rhino we would accept (in addition to the other two) a `DummyDotNetException` which is never really raised anywhere, so as good as ignored
Outside of Rhino any import error should always be one of `ImportError` and `SyntaxError` anyways

File: src/compas/plugins.py (hunk 2 above)
Comment (tomvanmele) [reply src/compas/plugins.py]: yes, why not `class DotNetException(BaseException): pass`

**Review by tomvanmele — APPROVED** [review]
LGTM, with one comment

File: src/compas/plugins.py (hunk 2 above)
Comment (chenkasirer) [reply src/compas/plugins.py]: no other reason except it never crossed my mind :)

**Pull request merged**